include/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp

```diff
@@ -0,0 +1,177 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem__
+#define __OpenSpaceToolkit_Astrodynamics_Flight_System_PropulsionSystem__
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+
+#include <OpenSpaceToolkit/Physics/Data/Direction.hpp>
+#include <OpenSpaceToolkit/Physics/Data/Scalar.hpp>
+#include <OpenSpaceToolkit/Physics/Unit.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Derived.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Time.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+namespace system
+{
+
+using ostk::core::types::Real;
+
+using ostk::physics::data::Direction;
+using ostk::physics::data::Scalar;
+using ostk::physics::units::Length;
+using ostk::physics::units::Mass;
+using ostk::physics::units::Time;
+using ostk::physics::units::ElectricCurrent;
+using ostk::physics::units::Angle;
+using ostk::physics::units::Derived;
+using ostk::physics::Unit;
+
+/// @brief                      Define a propulsion system (constant thrust, constant Isp for now)
+
+class PropulsionSystem
+{
+   public:
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     PropulsionSystem propulsion = { ... };
+    /// @endcode
+    ///
+    /// @param              [in] aThrust Thrust (scalar)
+    /// @param              [in] aSpecificImpulse Specific impulse
+
+    PropulsionSystem(const Scalar& aThrust, const Scalar& aSpecificImpulse);
```

Comment:
Is `Scalar` just like `Real`? (as in it has no units attached to it?)
If so, if this is a number, we might want to specify in the @param description the expected units at the end

Reply:
I just read further below that `Scalar` is `{value, unit}` in that case you can ignore my comment above.

Reply:
Yup exactly, it's value, unit